langchain/chains/llm_router_chain/base.py

```diff
@@ -0,0 +1,103 @@
+import re
+from types import FunctionType
+from typing import Dict, List, Any
+from pydantic import Extra, root_validator
+
+from langchain import LLMChain, BasePromptTemplate
+from langchain.chains.base import Chain
+from langchain.chains.conversation.prompt import PROMPT
+from langchain.input import get_color_mapping
+from langchain.memory import ConversationBufferWindowMemory
+from langchain.schema import BaseMemory
+
+
+class RouterChain(LLMChain):
```

Comment:
should inherit from Chain instead

Reply:
ok - maybe that'd unlock the issue with the output constraint.

Reply:
Took care of it - thanks. This helped me figure out what's required to maintain the `lastChain` context through history.